flake.nix

```diff
@@ -31,82 +38,48 @@
 
   # -------------------------------------------------------------------------- #
 
-  outputs = {
-    self,
-    nixpkgs,
-    sqlite3pp,
-    pre-commit-hooks,
-    crane,
-    fenix,
-    ...
-  } @ inputs: let
-    # Given a function `fn' which takes system names as an argument, produce an
-    # attribute set whose keys are system names, and values are the result of
-    # applying that system name to `fn'.
-    eachDefaultSystemMap = let
-      defaultSystems = [
-        "x86_64-linux"
-        "aarch64-linux"
-        "x86_64-darwin"
-        "aarch64-darwin"
-      ];
-    in
-      fn: let
-        proc = system: {
-          name = system;
-          value = fn system;
-        };
-      in
-        builtins.listToAttrs (map proc defaultSystems);
-
+  outputs = inputs: let
     # ------------------------------------------------------------------------ #
-
+    # Temporarily use nixpkgs-process-compose
+    nixpkgs.legacyPackages = {inherit (inputs.nixpkgs.legacyPackages) x86_64-linux x86_64-darwin aarch64-linux aarch64-darwin;};
+    nixpkgs.lib = inputs.nixpkgs.lib;
+  in rec {
     # Overlays
     # --------
+    overlays.deps = nixpkgs.lib.composeManyExtensions [
+      (final: prev: {
+        process-compose = final.callPackage (inputs.nixpkgs-process-compose + "/pkgs/applications/misc/process-compose") {};
```

Comment:
Did the package definition even change as much that we can't just override src/version in the stable Nixpkgs?

Reply:
Nope, only version and hash changes: https://github.com/NixOS/nixpkgs/commits/master/pkgs/applications/misc/process-compose/default.nix

Out of curiosity, does referencing [upstream's flake/package](https://github.com/F1bonacc1/process-compose/blob/main/flake.nix) at the appropriate tag with our own `nixpkgs` input at our own rev also achieve the same effect?